lib/admin/WP_Auth0_Admin_Advanced.php

```diff
@@ -407,11 +378,11 @@ public function basic_validation( array $old_options, array $input ) {
 	 * @return array
 	 */
 	public function migration_ws_validation( array $old_options, array $input ) {
-		$input['migration_ws']    = (int) ! empty( $input['migration_ws'] );
+		$input['migration_ws']    = $this->sanitize_switch_val( $input['migration_ws'] ?? null );
```

Comment:
on the `WP_Auth0_Options` file you changed existing `null` values to `''`. This line is setting one of those entries as null again. Is that intentional?

Reply:
Yes. This is doing null coalescing, so we're checking for the key (key will be missing for "off" values sent by the UI), then defaulting to `null`. The sanitize method there does the final conversion (starting with string type casting).